### PR TITLE
CP-2108 update poynt sample app - PART 1 - Order Service

### PR DESCRIPTION
**Pull request by artemkopan**
https://poyntc.atlassian.net/browse/CP-2108
What was done:
1. get orders
2. update order
3. get order
4. create order
5. create order return
6. get returns by order id

https://github.com/poynt/PoyntSamples/assets/7382939/faecddb5-e660-4839-9977-9cac38088d8e

### CODE REVIEW

File: commerce-services-sample/src/main/java/com/godaddy/commerce/services/sample/common/viewmodel/CommonViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+                    commonState.copy(loading = false, error = throwable.message)
+                )
+            }
+        }) {
+            execute()
+            update { copyCommonState(this::class, commonState.copy(loading = false, error = "")) }
+        }
+    }
+
+    // Do not use this method on real project.
```
Comment (flongo-godaddy) [inline commerce-services-sample/src/main/java/com/godaddy/commerce/services/sample/common/viewmodel/CommonViewModel.kt:43]: LOOOOOOL

**Review by flongo-godaddy — APPROVED** [review]
Databinding shinned in this PR

File: commerce-services-sample/src/main/java/com/godaddy/commerce/services/sample/common/viewmodel/CommonViewModel.kt (hunk 1 above)
Comment (lalkhawaja-godaddy) [reply commerce-services-sample/src/main/java/com/godaddy/commerce/services/sample/common/viewmodel/CommonViewModel.kt:43]: what they should use ?

Comment (artemkopan) [reply commerce-services-sample/src/main/java/com/godaddy/commerce/services/sample/common/viewmodel/CommonViewModel.kt:43]: I just update state in viewmodel. 
Each developer can decide what to use for their application

File: commerce-services-sample/build.gradle [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+    implementation "com.godaddy.commerce:sdk-common-ktx:$commerce_services"
+    implementation "com.godaddy.commerce:util-terminal:$commerce_services"
+    implementation "com.godaddy.commerce:provider-terminal:$commerce_services"
+
+    implementation "androidx.appcompat:appcompat:$appcompat"
+
+    implementation "androidx.navigation:navigation-fragment-ktx:$nav_version"
+    implementation "androidx.navigation:navigation-ui-ktx:$nav_version"
+    implementation "androidx.recyclerview:recyclerview:1.3.2"
```
Comment (dshuba-goddady) [inline commerce-services-sample/build.gradle:61]: Can we use versions.toml file for dependencies version control?

File: commerce-services-sample/src/main/java/com/godaddy/commerce/services/sample/orders/OrderViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+                service.getOrders(bundle, it.onSuccess(), it.onError())
+            }
+            update { copy(items = response.mapToUiItems(::returnOrder)) }
+        }
+    }
+
+    private fun returnOrder(order: Order) {
+        execute {
+            val lineItemReturns = requireNotNull(order.lineItems) {
+                "Nothing to return. Line items are empty"
```
Comment (dshuba-goddady) [inline commerce-services-sample/src/main/java/com/godaddy/commerce/services/sample/orders/OrderViewModel.kt:73]: Can we have an edge case when `lineItems` it not null, but size is 0?

**Review by dshuba-goddady — APPROVED** [review]

File: commerce-services-sample/build.gradle (hunk 2 above)
Comment (artemkopan) [reply commerce-services-sample/build.gradle:61]: we can't due to gradle version 4

File: commerce-services-sample/src/main/java/com/godaddy/commerce/services/sample/orders/OrderViewModel.kt (hunk 3 above)
Comment (artemkopan) [reply commerce-services-sample/src/main/java/com/godaddy/commerce/services/sample/orders/OrderViewModel.kt:73]: I don't think that we can create order without line items

**Review by mvideshyn-godaddy — APPROVED** [review]

**Review by ylutsenko-godaddy — APPROVED** [review]

File: commerce-services-sample/build.gradle (hunk 2 above)
Comment (dshuba-goddady) [reply commerce-services-sample/build.gradle:61]: Gotcha, thanks for context

**Review by dshuba-goddady — APPROVED** [review]

**Pull request merged**